apps/taxes/src/modules/providers-configuration/providers-config-input-schema.ts

```diff
@@ -0,0 +1,15 @@
+import { z } from "zod";
+import { providerSchema } from "./providers-config";
+
+export const updateProviderInstanceInputSchema = z.object({
+  id: z.string(),
+  provider: providerSchema,
+});
+
+export const deleteProviderInstanceInputSchema = z.object({
+  id: z.string(),
+});
+
+export const createProviderInstanceInputSchema = z.object({
+  provider: providerSchema,
```

Comment:
Do we need nested objects? Maybe flattening the input would be a better choice?

Reply:
I wanted to retain the same format across all the inputs. Some need `provider`, some need `id`, but some need both:

```
export const updateProviderInstanceInputSchema = z.object({
  id: z.string(),
  provider: providerSchema,
});

export const deleteProviderInstanceInputSchema = z.object({
  id: z.string(),
});

export const createProviderInstanceInputSchema = z.object({
  provider: providerSchema,
});
```

Merging those two is not desired, unfortunately.